File: javascript/clock/clock.js
export class Clock {
  constructor(hours = 0, minutes = 0) {
    this.minutes = 60 * hours + minutes;
  }

  toString() {
    let minutes = ((this.minutes % 60) + 60) % 60
    let hours = ((Math.floor(this.minutes / 60) % 24) + 24) % 24;
    return `${('0' + hours).slice(-2)}:${('0' + minutes).slice(-2)}`;
  }

  plus(minutes) {
    return new Clock(0, this.minutes + minutes);
  }

  minus(minutes) {
    return new Clock(0, this.minutes - minutes);
  }

  equals(clock2) {
    return this.toString() === clock2.toString();
  }
}

File: java/clock/src/main/java/Clock.java
class Clock {

    private int minutes;

    Clock(int hours, int minutes) {
        this.minutes = (((60 * hours + minutes + 1440) % 1440) + 1440) % 1440;
    }

    void add(int minutes) {
        this.minutes = (((this.minutes + minutes) % 1440) + 1440) % 1440;
    }

    @Override
    public String toString() {
        return String.format("%02d:%02d", this.minutes / 60, this.minutes % 60);
    }

    @Override
    public boolean equals(Object obj) {
        if (obj == this) return true;
        if (obj == null) return false;
        if (getClass() != obj.getClass()) return false;
        Clock clock = (Clock) obj;
        return this.minutes == clock.minutes;
    }

    @Override
    public int hashCode() {
        return this.minutes;
    }

}